lib/chef/application/client.rb

```diff
@@ -454,11 +460,12 @@ def interval_run_chef_client
   end
 
   def sleep_then_run_chef_client(sleep_sec)
-    @signal = test_signal
-    unless @signal == IMMEDIATE_RUN_SIGNAL
-      Chef::Log.debug("Sleeping for #{sleep_sec} seconds")
-      interval_sleep(sleep_sec)
-    end
```

Comment:
We didn't use `@signal` anywhere else so I moved all the signal code here into `interval_sleep` and added an explicit line for `IMMEDIATE_RUN` for clarity.